src/robot/conf/languages.py

```diff
@@ -67,36 +71,57 @@ def _get_languages(self, languages):
                 if normalized in available:
                     returned.append(available[normalized]())
                 else:
-                    returned.extend(self._import_languages(lang))
+                    returned.extend(v() for v in self.import_languages_module(lang).values())
         return returned
 
-    def _resolve_languages(self, languages):
+    def _resolve_languages(self, languages, add_default=True):
         if not languages:
             languages = []
         elif is_list_like(languages):
             languages = list(languages)
         else:
             languages = [languages]
-        languages.append(En())
+        if add_default:
+            languages.append(En())
+            # The English singular forms are added for backwards compatibility
+            self.headers = {
+                'Setting': 'Settings',
+                'Variable': 'Variables',
+                'Test Case': 'Test Cases',
+                'Task': 'Tasks',
+                'Keyword': 'Keywords',
+                'Comment': 'Comments'
+            }
         return languages
 
-    def _get_available_languages(self):
+    @staticmethod
+    def get_available_languages():
+        """Returns the currently available Languages as a dict of names and classes."""
         available = {}
         for lang in Language.__subclasses__():
             available[normalize(lang.__name__)] = lang
             if lang.__doc__:
                 available[normalize(lang.__doc__.splitlines()[0])] = lang
         return available
 
-    def _import_languages(self, lang):
+    @staticmethod
+    def import_languages_module(file_or_module):
+        """Imports a custom language module and returns the available languages."""
         def is_language(member):
             return (inspect.isclass(member)
                     and issubclass(member, Language)
                     and member is not Language)
-        if os.path.exists(lang):
-            lang = os.path.abspath(lang)
-        module = Importer('language file').import_module(lang)
-        return [value() for _, value in inspect.getmembers(module, is_language)]
+
+        if os.path.exists(file_or_module):
+            file_or_module = os.path.abspath(file_or_module)
+        file_or_module = Importer('language file').import_module(file_or_module)
```

Comment:
Would `path_or_name` be a better name than `file_or_module`? `file` often refers to a open file object that we don't support. `module` could also refer to an actual module object when we actually except module name.

The value assigned above should be `module` because it is an actual imported module.